src/Sulu/Bundle/ContactBundle/Controller/AbstractMediaController.php

```diff
@@ -206,50 +222,119 @@ protected function removeMediaFromEntity($entityName, $id, $mediaId)
     protected function getMultipleView($entityName, $routeName, AbstractContactManager $contactManager, $contactId, $request)
     {
         try {
+            /** @var UserInterface $user */
+            $user = $this->getUser();
             $locale = $this->getUser()->getLocale();
 
             if ('true' === $request->get('flat')) {
-                $listBuilder = $this->listBuilderFactory->create($entityName);
-                $fieldDescriptors = $this->getFieldDescriptors($entityName, $contactId);
-                $listBuilder->setIdField($fieldDescriptors['id']);
-                $this->restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
-
-                $listResponse = $listBuilder->execute();
-                $listResponse = $this->addThumbnails($listResponse, $locale);
-                $listResponse = $this->addUrls($listResponse, $locale);
-
-                $list = new ListRepresentation(
-                    $listResponse,
-                    static::$mediaEntityKey,
-                    $routeName,
-                    \array_merge(['contactId' => $contactId], $request->query->all()),
-                    $listBuilder->getCurrentPage(),
-                    $listBuilder->getLimit(),
-                    $listBuilder->count()
-                );
+                if (null === $this->mediaListRepresentationFactory) {
+                    $listRepresentation = $this->legacyGetMultipleView(
+                        $entityName,
+                        $routeName,
+                        $contactId,
+                        $request,
+                        $locale
+                    );
+                } else {
+                    $fieldDescriptors = $this->mediaListRepresentationFactory->getFieldDescriptors();
+
+                    $fieldDescriptors['contactId'] = new DoctrineFieldDescriptor(
```

Comment:
`contactId` is just a name and represents the id of either a contact or an account. It also was like that before. Feel free to suggest a more generic name which is good for both, contacts and accounts. But since both, the `AccountManager` and the `ContactManager` extends from the `AbstractContactManager`, I don't think this naming is too bad here.

Reply:
I am not happy with the contact and account naming all together, especially since they are called differently in the UI :see_no_evil: But if it was like that before, I would say we ignore that issue for this PR.